src/main/java/hudson/os/windows/ManagedWindowsServiceLauncher.java

```diff
@@ -385,24 +387,26 @@ private String resolveJava(SlaveComputer computer) {
 
     // -- duplicates code from ssh-slaves-plugin
     private EnvVars getEnvVars(SlaveComputer computer) {
-        final EnvVars global = getEnvVars(Jenkins.getInstance());
-
-        final EnvVars local = getEnvVars(computer.getNode());
-
-        if (global != null) {
-            if (local != null) {
-                final EnvVars merged = new EnvVars(global);
-                merged.overrideAll(local);
-
-                return merged;
-            } else {
-                return global;
+        Slave node = computer.getNode();
+        final EnvVars local = node != null ? getEnvVars(node) : null;
+        
+        Jenkins jenkins = Jenkins.getInstance();
+        if (jenkins != null) {
```

Comment:
:ant: just throw an exception otherwise. Obviously we are not launching any agents after Jenkins has shut down.